clusterloader2/pkg/measurement/common/generic_query_measurement.go

```diff
@@ -68,39 +104,58 @@ func (g *genericQueryGatherer) IsEnabled(config *measurement.Config) bool {
 }
 
 func (g *genericQueryGatherer) Gather(executor QueryExecutor, startTime, endTime time.Time, config *measurement.Config) ([]measurement.Summary, error) {
-	metric, err := g.query(executor, startTime, endTime)
+	var errs []error
+	data := map[string]float64{}
+	for _, q := range g.queries {
+		samples, err := g.query(q, executor, startTime, endTime)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(samples) != 1 {
```

Comment:
I'm wondering if also len(samples) == 0 should be accepted.

Let's take a case like I want to track latency of POST on pods, so I write query like:
```
        - name: Perc99
          query: histogram_quantile(0.99, sum(rate(apiserver_request_duration_seconds_bucket{verb="POST", resource="pods}[%v])) by (le))
          threshold: 60
```

Let's say that there are no post on pods. I would expect this to not fail (as there is no sample violating the theshold).

For summary/perfdash you can log any value you want (e.g. not create a key for a query with 0 samples or log 0.0 or NaN).

Reply:
not creating a key for a query in case of 0 samples seems to be the cleanest solution IMHO

Reply:
Added skipping queries that return no results